File: frontend/src/utils/Ticket.js
export default class Ticket {
    constructor() {
        this.sender = "Frontend"
        this.receiver = undefined;
        this.type = undefined;
        this.data = undefined;
    }

    fromObject(newTicket) {
        this.receiver = newTicket.receiver;
        this.type = newTicket.type;
        this.data = newTicket.data;

        return JSON.stringify({
            "sender": this.sender,
            "receiver": this.receiver,
            "type": this.type,
            "data": this.data
        })
    }
}

File: backtester/utils/ticket.py
import json

class Ticket:
    def __init__(self) -> None:
        self.sender = "Backtester"
        self.receiver = None
        self.type = None
        self.data = None

    def from_object(self, newTicket):
        self.receiver = newTicket.get('receiver', self.receiver)
        self.type = newTicket.get('type', self.type)
        self.data = newTicket.get('data', self.data)

        return json.dumps({
            'sender': self.sender,
            'receiver': self.receiver,
            'type': self.type,
            'data': self.data
        }, allow_nan=True)
